src/Compilers/Core/Portable/PEWriter/DebugSourceDocumentLocationComparer.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Cci
+{
+    internal sealed class DebugSourceDocumentLocationComparer : IComparer<DebugSourceDocument>
+    {
+        public static DebugSourceDocumentLocationComparer Instance
+            => new DebugSourceDocumentLocationComparer();
+
+        private DebugSourceDocumentLocationComparer()
+        {
+        }
+
+        int IComparer<DebugSourceDocument>.Compare(DebugSourceDocument x, DebugSourceDocument y)
+            => x.Location.CompareTo(y.Location);
```

Comment:
>x.Location.CompareTo(y.Location) [](start = 15, length = 32)

It looks like Location is a string, we should be very explicit what kind of comparison we would like to perform. For example it looks like CompareTo performs a word (case-sensitive and culture-sensitive) comparison using the current culture. I am not sure this is the kind of comparison we would want to use.